tests/conftest.py

```diff
@@ -136,6 +136,23 @@ def assert_success_and_collect(function, params, url="https://localhost/", metho
         finally:
             httpretty.disable()
 
+    @staticmethod
+    def assert_404_and_collect(function, params, url="https://localhost/", method="GET"):
```

Comment:
The amount of copy-paste within this fixture is beyond reasonable threshold.
Please refactor it.